lib/fiddle/jruby.rb

```diff
@@ -158,6 +158,17 @@ def call(*args, &block);
           end
           args[i] = Fiddle::JRuby.__ffi_type__(args[i])
         end
+      else
+        args.size.times do |i|
+          arg = args[i]
+          next unless arg.respond_to?(:to_ptr)
+          loop do
+            arg = arg.to_ptr
+            break if arg.is_a?(FFI::Pointer)
+            break unless arg.respond_to?(:to_ptr)
+          end
+          args[i] = arg
+        end
```

Comment:
Interesting that this is needed on TruffleRuby but apparently not on JRuby.
I'll repro to see what's the difference there.

Reply:
OK it's needed, details explained in https://github.com/ruby/fiddle/pull/149/commits/0631d46cddb4dc64ff449ddc62524c6767e9b66b